src/version.ts

```diff
@@ -0,0 +1,318 @@
+  // Occasionally the connection is reset for unknown reasons
+  // In those cases, retry the download
+  const versionsFile = await retry(
+    async () => {
+      const response = await fetch(VERSIONS_JSON_URL)
+      return response.text()
+    },
+    {
+      retries: 5,
+      onRetry: (err: Error) => {
+        core.info(
+          `Download of versions.json failed, trying again. Error: ${err}`
+        )
+      },
+    }
+  )
+
+  if (!versionsFile) {
+    throw new Error(`Unable to download versions.json after 5 attempts`)
+  }
+
+  return JSON.parse(versionsFile)
+}
+
+/**
+ * Determine the latest Julia release associated with the version range
+ * (e.g. "1", "^1.2.3", "~1.2.3"). Additionally, supports the version aliases:
+ *
+ * - `lts`: The latest released long-term stable (LTS) version of Julia.
+ * - `min`: The earliest version of Julia within the `juliaCompatRange`.
+ *
+ * @param versionRange: The node version range or alias.
+ * @param availableVersions: An array of available Julia versions.
+ * @param includePrereleases: Allow prereleases to be used when determining
+ * the version number.
+ * @param juliaCompatRange: The Node semver range to further restrict the results
+ * @returns The full semver version number
+ * @throws Error if the version specifier doesn't overlap with any available
+ * Julia releases.
+ */
+export function resolveVersion(
+  versionRange: string,
+  availableVersions: string[],
+  juliaCompatRange: string | null = null
+): string | null {
+  if (
+    semver.valid(versionRange) == versionRange &&
+    availableVersions.includes(versionRange)
+  ) {
+    // versionRange is already a valid semver version (not a semver range)
+    return versionRange
+  } else if (versionRange === "min") {
+    // Resolve "min" to the minimum supported Julia version compatible with the
+    // project file
+    if (!juliaCompatRange) {
+      throw new Error(
+        'Unable to use version "min" when the Julia project file does not specify a compat for Julia'
+      )
+    }
+    return semver.minSatisfying(availableVersions, juliaCompatRange)
+  } else if (versionRange === "lts") {
+    return semver.maxSatisfying(availableVersions, LTS_VERSION)
+  } else {
+    // Use the highest available version that match the version range
+    return semver.maxSatisfying(availableVersions, versionRange)
+  }
+}
+
+export function getNightlyUrl(
+  nightly: NightlyPlatform,
+  majorMinorVersion: string | null = null
+): string {
+  const majorMinorDir = majorMinorVersion ? majorMinorVersion + "/" : ""
+  const suffix = nightly.suffix ?? `${nightly.platform}-${nightly.arch}`
+  const url = `${NIGHTLY_BASE_URL}/${nightly.platform}/${nightly.arch}/${majorMinorDir}julia-latest-${suffix}.${nightly.ext}`
+  return url
+}
+
+async function urlExists(url: string): Promise<boolean> {
+  // Perform a HEAD request to validate the HTTP server contains the specified file exists.
+  // TODO: Verify the performance of these requests in CI.
```

Comment:
Lingering TODO, there's another within `genNightlies()`

Reply:
When confirming `1.12-nightly` I saw 9s on the initial check and 1s on follow up checks.